server/game.py

```diff
@@ -215,7 +215,13 @@ def update_passengers_count(self):
         """Update the number of passengers based on the number of trains"""
         # Calculate the desired number of passengers based on the number of alive trains
         self.desired_passengers = (
-            len([train for train in self.trains.values() if train.alive])
+            len(
+                [
+                    train
+                    for train in self.trains.values()
+                    if self.contains_train(train.nickname)
```

Comment:
nit: students haven't seen this form of list comprehension (with the `if` filter).